packages/lodestar/src/sync/sync.ts

```diff
@@ -246,7 +247,22 @@ export class BeaconSync implements IBeaconSync {
       }
       retry++;
     } // end while
-    this.processingRoots.delete(missingRootHex);
     if (!found) this.logger.error("Failed to get unknown ancestor root", {blockRoot: missingRootHex});
   };
+
+  /**
+   * Check if we should process a missing root.
+   * Time-cache does not do the sweep() automatically,
+   * it only does that per put() call.
+   */
+  private shouldProcessRoot(missingRootHex: string): boolean {
+    if (this.processingRoots.has(missingRootHex)) {
+      // time-cache only do sweep() in this case
+      this.processingRoots.put(missingRootHex);
+      // check again to see if the item is there after the sweep()
+      return !this.processingRoots.has(missingRootHex);
```

Comment:
this is the sweep() implementation: https://github.com/daviddias/time-cache/blob/master/src/index.js#L27